src/sentry/plugins/providers/repository.py

```diff
@@ -65,14 +66,34 @@ def dispatch(self, request, organization, **kwargs):
                 }, status=400
             )
 
-        repo = Repository.objects.create(
-            organization_id=organization.id,
-            name=result['name'],
-            external_id=result.get('external_id'),
-            url=result.get('url'),
-            config=result.get('config') or {},
-            provider=self.id,
-        )
+        try:
+            with transaction.atomic():
+                repo = Repository.objects.create(
+                    organization_id=organization.id,
+                    name=result['name'],
+                    external_id=result.get('external_id'),
+                    url=result.get('url'),
+                    config=result.get('config') or {},
+                    provider=self.id,
+                )
+        except IntegrityError:
+            # Try to delete webhook we just created
+            try:
+                repo = Repository(
+                    organization_id=organization.id,
+                    name=result['name'],
+                    external_id=result.get('external_id'),
+                    url=result.get('url'),
+                    config=result.get('config') or {},
+                    provider=self.id,
+                )
+                self.delete_repository(repo, actor=request.user)
```

Comment:
@MaxBittker kind of confusing, but this `self.delete_repository` doesn't actually delete anything from the database, it just deletes the webhook that we would have already created in github/bitbucket (see line 55 above: https://github.com/getsentry/sentry/pull/6411/files#diff-7158a0c29ea647b04fbf0562adb8baccR55)

Reply:
oh!